lib/libc/riscv/SYS.h

```diff
@@ -42,22 +42,46 @@
 	li	t0, SYS_ ## name;				\
 	ecall
 
+#ifndef __CHERI_PURE_CAPABILITY__
+#define _GET_LOCAL_FNPTR(outreg, function)	\
+	lla outreg, _C_LABEL(function)
+#define _GET_FNPTR(outreg, function)	\
+	la outreg, _C_LABEL(function)
+#define _CALL_FNPTR(reg)	jalr cra, reg
+#define _TAILCALL_FNPTR(reg)	jr reg
+#define _RETURN	ret
+#else
+#define _GET_FNPTR(outreg, function)	\
+	clgc CAPABILITY_REG(outreg), _C_LABEL(function)
+#define _GET_LOCAL_FNPTR(outreg, function)	\
+	cllc CAPABILITY_REG(outreg), _C_LABEL(function)
+#define _CALL_FNPTR(reg)	cjalr cra, CAPABILITY_REG(reg)
+#define _TAILCALL_FNPTR(reg)	cjr CAPABILITY_REG(reg)
+#define _RETURN	cret
+#endif
+
+#define ASM_LOCAL_TAILCALL(function)				\
```

Comment:
I would probably pass the register to these so it's clearer to the reader which register is being used as a temporary.